engine/src/main/java/org/destinationsol/game/SolObject.java

```diff
@@ -16,41 +16,187 @@
+     * when it gets too far from player/camera, and instead of recreating from {@code FarObject}, to be generated again.
+     *
+     * @return FarObject representation of this object, or null if object is to be removed.
+     */
     FarObject toFarObject();
 
+    /**
+     * Returns list of all {@link Drawable Drawables} this object has assigned.
+     *
+     * @return List of drawables this object has.
+     */
     List<Drawable> getDrawables();
 
+    /**
+     * Returns object's angle.
+     * <p>
+     * This method usually just returns the angle retrieved from object's internal {@link Body}.
+     *
+     * @return Angle of this object, in degrees.
+     */
     float getAngle();
 
+    /**
+     * Returns object's speed.
+     * <p>
+     * This method usually just returns the vector retrieved from object's internal {@link Body}.
+     *
+     * @return Speed of this object.
+     */
     Vector2 getSpeed();
 
+    /**
+     * Called whenever this object touches another object, and can be used for custom collisions.
+     * <p>
+     * This method is for instance used for collecting {@link Loot} by ships when these two touch. This method is not to
+     * be used for handling of applying force between the two objects nor playing any collision sounds, unless the
+     * object is specifically created to do so. This method should however apply collision damage. This method can be
+     * freely left blank if there is no custom contact handling.
+     *
+     * @param other      Object this object touches.
+     * @param absImpulse Impulse the objects apply to each other.
+     * @param game       Game this object belongs to.
+     * @param collPos    Position where the two objects touch.
+     */
     void handleContact(SolObject other, float absImpulse, SolGame game, Vector2 collPos);
 
+    /**
+     * Used for retrieval of object's debug string.
+     * <p>
+     * Any kind of information can be used in the debug string, if you don't need/want to display any debug information,
+     * you can freely have this method return null. To display debug strings in-game, set the flag {@link DebugOptions#OBJ_INFO}.
+     * These strings should then be rendered in the proximity of their objects.
+     *
+     * @return Debug string with information about the object.
+     */
     default String toDebugString() {
         return null;
     }
 
+    /**
+     * Denotes whether this object is meant to behave as if made from metallic material.
+     * <p>
+     * This method is used mainly for choosing the sound to play after collision. Value true will play metallic sound,
+     * false will play rocky sound. Value of null means no sound will be played.
+     *
+     * @return Whether this object is made from metal.
+     */
+    //TODO allow for more complex collision sounds. Or even better, completely rework the material system.
```

Comment:
Unless I am missing something, this TODO item may belong to an implementor rather than the interface itself.

Reply:
There is about 10? 15? implementors of this interface. All of them should have better possibility to choose the sound to be played upon crashing with something.

Reply:
This still looks like something that should belong to a specific implementor, since an interface contains absolutely no implementation information, and as such a TODO item regarding interface implementation should in my opinion not belong to the interface. Also, CodeFactor (rightly) complains that this comment is unexpected here. 

Anyhow, I don't see this as too big of a problem. It might make sense to move this comment to a better place so that CodeFactor stops complaining. Maybe move it _before_ the JavaDoc comment? I don't know if that will work, but it is worth a try. If nothing works, then we can leave it as is.

Reply:
CodeFactor complaints of ANY //TODO

Reply:
I honestly do not think that is the case here -- the placement of the comment is causing the problem, not the fact that it is a TODO comment. Still, I see no point discussing this further, and leave it to your discretion to act further.

Reply:
I just realized that CodeFactor does complain of _any_ TODO it sees! That must have been the problem here.

Reply:
Told ya :smile:

Reply:
Yep, sorry for doubting you! :D